homeassistant/components/light/decora_wifi.py

```diff
@@ -0,0 +1,245 @@
+"""
+Interfaces with the myLeviton API for Decora Smart WiFi products.
+
+See:
+http://www.leviton.com/en/products/lighting-controls/decora-smart-with-wifi
+
+Uses Leviton's cloud services API for cloud-to-cloud integration.
+
+"""
+
+import json
+import logging
+import requests
+
+import voluptuous as vol
+
+from homeassistant.components.light import (ATTR_BRIGHTNESS,
+                                            ATTR_TRANSITION,
+                                            Light,
+                                            PLATFORM_SCHEMA,
+                                            SUPPORT_BRIGHTNESS,
+                                            SUPPORT_TRANSITION)
+from homeassistant.const import (CONF_USERNAME, CONF_PASSWORD,
+                                 EVENT_HOMEASSISTANT_STOP)
+import homeassistant.helpers.config_validation as cv
+import homeassistant.loader as loader
+
+REQUIREMENTS = []
+
+_LOGGER = logging.getLogger(__name__)
+
+# Validation of the user's configuration
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+})
+
+"""
+The myLeviton API also supports some sort of socket communication...
+Probably better for the polling/update comms, but I haven't really looked
+at it.
+"""
+LEVITON_ROOT = 'https://my.leviton.com/api'
+DOMAIN = 'myLeviton'
```

Comment:
Or you can remove it, you need no DOMAIN on platform, for hass.data you can use `DATA_XY`